sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -131,6 +129,9 @@ def _set_partition_key(
             return _return_undefined_or_empty_partition_key(self.is_system_key)
         return cast(Union[str, int, float, bool, List[Union[str, int, float, bool]]], partition_key)
 
+    def __get_client_container_caches(self):
+        return self.client_connection._container_properties_cache  # pylint: disable=protected-access
```

Comment:
How often is this value/properties likely to change? Given that we're storing this on the connection, there's the potential for race conditions in concurrent scenarios. Though this may not be a problem if the worst case scenario is that two threads both attempt to update the cache at once and both retrieve the same data.
Do the properties of a container change?

Reply:
For that reason we only store properties that will not change in the life time of the container such as partition key definition, the container link, and the container RID. Other properties will need to be grabbed through a container read request as they are prone to change.  The only time when the cached properties of a specific container would change is due to container recreate.